From linux-kernel-owner@vger.kernel.org Sun Apr 16 19:04:07 2006
From: Jesper Juhl <jesper.juhl@gmail.com>
To: linux-kernel@vger.kernel.org
Subject: PCI: fix potential resource leak in drivers/pci/msi.c
Date: Mon, 17 Apr 2006 04:02:54 +0200
Cc: Tom Long Nguyen <tom.l.nguyen@intel.com>, Jesper Juhl <jesper.juhl@gmail.com>
Content-Disposition: inline
Message-Id: <200604170402.54458.jesper.juhl@gmail.com>

The coverity checker spotted (as entry #599) that we might leak `entry' in 
drivers/pci/msi.c::msix_capability_init()
This patch should take care of that.


Signed-off-by: Jesper Juhl <jesper.juhl@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/pci/msi.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- gregkh-2.6.orig/drivers/pci/msi.c
+++ gregkh-2.6/drivers/pci/msi.c
@@ -793,8 +793,10 @@ static int msix_capability_init(struct p
 		if (!entry)
 			break;
 		vector = get_msi_vector(dev);
-		if (vector < 0)
+		if (vector < 0) {
+			kmem_cache_free(msi_cachep, entry);
 			break;
+		}
 
  		j = entries[i].entry;
  		entries[i].vector = vector;
